### PR TITLE
Fix GCSFS read API and added test.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/utils/gcsfs.py [hunk 1]
```diff
@@ -235,7 +259,9 @@ def generic_write(output_string, output_path):
 
 
 def generic_read(path):
-  """Reads the whole content of the provided location. Has to be a GCS path or
+  """Reads the whole content of the provided location.
+
+  Has to be a GCS path or
```
Comment (jysohn23) [inline torch_xla/utils/gcsfs.py]: Nit: Line breaking looks weird.

Comment (dlibenzi) [reply torch_xla/utils/gcsfs.py]: Duh! `google-pyformat` change.

File: torch_xla/utils/gcsfs.py [hunk 2]
```diff
@@ -220,6 +243,7 @@ def write(path, content):
 
 def generic_write(output_string, output_path):
   """Write a string/bytes or file into a GCS blob or local disk, depending
+
```
Comment (taylanbil) [inline torch_xla/utils/gcsfs.py]: nit: newline not needed.

Comment (dlibenzi) [reply torch_xla/utils/gcsfs.py]: Actually, it is.
The comment is wrong. Head comment must be a one liner, so google-pyformat tried to fix that.

Comment (dlibenzi) [reply torch_xla/utils/gcsfs.py]: Fixed this.

**Pull request merged**